File: CS471/hw1/ack.py
def ackermann(M, N):
    if(M == 0):
        return(N+1)
    elif( (N == 0) and (M>0) ):
        return( ackermann(M-1, 1) )
    elif( (N>0) and (M>0) ):
        return( ackermann(M-1, ackermann(M, N-1) ) )

def main():
    N = input("Please enter an integer number: ")
    print("Your number (N) is: %d\n" % N)
    print("This is the ackerman number for ack(3, N): %d\n" % ackermann(3,N))

main()
i = 10
print("This is a big mistake %d" % i)

File: CS471/hw1/ack.rb

def ack(m,n)
    if m == 0
        n+1
    elsif n == 0 and m > 0
        ack(m-1, 1)
    elsif n > 0 and m > 0
        ack(m-1, ack(m, n-1) )
    end
end

def main()
    puts "Please enter a single integer: "
    n = gets.to_i
    
    puts "Your inputted number (n) is: " + n.to_s
    result = ack(3,n)
    puts "This is the ackerman number for ack(3, N): " + result.to_s
    
end

main()
